### PR TITLE
Bump to version 1.18.0 and require Elixir ~> 1.7

### PR DESCRIPTION
**Pull request by fatcatt316**
## What Changed?
- [x] Require Elixir ~. 1.7
- [x] Upgrade ex_doc from 

## Why?

Requiring supported version of Elixir.

## Tests
```
mix test
4 doctests, 47 tests, 0 failures, 1 excluded
```

```
mix test --only e2e
4 doctests, 47 tests, 0 failures, 50 excluded
```

## Questions for Reviewers
1. Any other upgrades that'd be worth doing for this version?
2. Would it be better to _just_ have "Spreedly" as the maintainer?
3. Any other tests you'd recommend doing?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,28 @@
 # Kaffe
 
-An opinionated, highly specific, Elixir wrapper around
-[Brod](https://github.com/klarna/brod): the Erlang Kafka client.
-:coffee:
+An opinionated, highly specific, Elixir wrapper around [Brod](https://github.com/klarna/brod): the Erlang Kafka client. :coffee:
```
Comment (fatcatt316) [inline README.md:3]: Most of the changes in the README, besides adding the table of contents, are just unwrapping lines.

File: mix.exs [hunk 2]
```diff
@@ -28,15 +28,15 @@ defmodule Kaffe.Mixfile do
...
+      {:ex_doc, "~> 0.20", only: :dev, runtime: false},
     ]
   end
 
   defp package do
     [
       name: :kaffe,
       licenses: ["MIT License"],
-      maintainers: ["Kevin Lewis", "David Santoso", "Ryan Daigle", "Spreedly"],
+      maintainers: ["Kevin Lewis", "Spreedly", "Joe Peck"],
```
Comment (jknipp) [inline mix.exs:39]: 🤔 I'm on the fence if we should remove maintainers on an open-source project. Theoretical they could still contribute?

Also, we'll want to publish this release to Hex. Looks like @objectuser was the last to do so. https://hex.pm/packages/kaffe

**Comment by jknipp** [comment]
@fatcatt316 We'll want to tag this after we merge and create a Github release with a short release note to indicate this release supports only Elixir 1.7 and above.

File: mix.exs (hunk 2 above)
Comment (fatcatt316) [reply mix.exs]: @jknipp Ah, good point. I added them back, since it's open-source.

File: mix.exs [hunk 3]
```diff
@@ -4,13 +4,13 @@ defmodule Kaffe.Mixfile do
...
       app: :kaffe,
-      version: "1.17.0",
+      version: "1.18.0",
       description:
         "An opinionated Elixir wrapper around brod, the Erlang Kafka client, that supports encrypted connections to Heroku Kafka out of the box.",
       name: "Kaffe",
       source_url: "https://github.com/spreedly/kaffe",
       package: package(),
-      elixir: "~> 1.3",
+      elixir: "~> 1.7",
```
Comment (objectuser) [inline mix.exs:13]: What's the semver implication of this? Does this imply needing to bump a major version?

Comment (jknipp) [reply mix.exs:13]: Not sure we need a major bump. A minor bump will still allow older runtimes to pin to `kaffe ~> 1.17.0`.

Comment (fatcatt316) [reply mix.exs:13]: @objectuser Good question! At first I agreed, but then read [this article](https://medium.com/@sampart/semantic-versioning-when-you-change-the-required-programming-language-version-16a3a3555c95), which seems to fit what we're doing here. They end up settling for bumping the minor version. A penny for your thoughts 💰

Comment (objectuser) [reply mix.exs:13]: I suppose it's a reasonable argument. I don't have a better one. 👍

**Comment by fatcatt316** [comment]
**Tagged version `1.18.0`**
https://github.com/spreedly/kaffe/releases/tag/1.18.0

**Hex release:**
_in progress_

**Comment by fatcatt316** [comment]
I intend to publish this new version to Hex by following these instructions: https://hex.pm/docs/publish

Lemme know if there's some different instructions you'd recommend!

**Comment by objectuser** [comment]
I bet I need to give you permissions to publish ...

**Comment by fatcatt316** [comment]
@objectuser You right

```
Publishing package...
[#########################] 100%
account not authorized for this action
Publishing failed
```

 \*sad trombone  sound*

**Comment by objectuser** [comment]
> The current authenticated user will be the package owner. Only package owners can publish the package, new owners can be added with the mix hex.owner task.

Same username on `hex.pm`?

**Pull request merged**